### PR TITLE
Add additional null checks to VPAs

### PR DESCRIPTION
**Pull request by JLineaweaver**
### What does this PR do?
We're seeing occasional panics when running on a larger set of clusters. Add additional null checks to make sure our agent doesn't panic

### Motivation
Want to fix any panics our agent might encounter

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by fisherevans — APPROVED** [review]
looks nil safer

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/verticalpodautoscaler.go [hunk 1]
```diff
@@ -29,6 +33,10 @@ func ExtractVerticalPodAutoscaler(v *v1.VerticalPodAutoscaler) *model.VerticalPo
 // extractVerticalPodAutoscalerSpec converts the Kubernetes spec to our custom one
 // https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L73
 func extractVerticalPodAutoscalerSpec(s *v1.VerticalPodAutoscalerSpec) *model.VerticalPodAutoscalerSpec {
+	if s == nil {
```
Comment (mfpierre) [inline pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/verticalpodautoscaler.go:36]: AFAICT this one should always be there https://github.com/kubernetes/autoscaler/blob/ab2e467452d7786a3cdb655bd26f220eaa8270a0/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L58 (not optional)

**Review by mfpierre — CHANGES_REQUESTED** [review]

File: pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/verticalpodautoscaler.go (hunk 1 above)
Comment (JLineaweaver) [reply pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/verticalpodautoscaler.go:36]: Do you think this should be removed to match the structure? Is there any harm in additional nil checks?

Comment (JLineaweaver) [reply pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/verticalpodautoscaler.go]: [106fdd9](https://github.com/DataDog/datadog-agent/pull/15488/commits/106fdd9fd3c5cdb59ed41817e2ff346bd3319168)

Comment (kangyili) [reply pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/verticalpodautoscaler.go]: > Is there any harm in additional nil checks?

IMO, without nil check, we can find issue earlier if we have nil spec while spec shouldn't be nil

**Pull request closed without merging**